src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/SharedUtils.java

```diff
@@ -382,20 +391,6 @@ static MethodHandle wrapWithAllocator(MethodHandle specializedHandle,
 
         closer = collectArguments(closer, insertPos++, MH_CLOSE_CONTEXT); // (Throwable, V?, Addressable?, BindingContext) -> V/void
 
-        if (!upcall) {
```

Comment:
This can go away, because we use acquire/release anyway to keep these args alive in the new adapter/wrapper.